src/core/models.py

```diff
@@ -1919,3 +1968,641 @@ def log_hijack_ended(sender, hijacker, hijacked, request, **kwargs):
 
 hijack_started.connect(log_hijack_started)
 hijack_ended.connect(log_hijack_ended)
+
+
+class OrganizationName(models.Model):
+    value = models.CharField(
+        max_length=1000,
+        verbose_name="Organization name",
+    )
+    ror_display_for = models.OneToOneField(
+        'Organization',
+        on_delete=models.CASCADE,
+        related_name='ror_display',
+        blank=True,
+        null=True,
+        help_text="This name is a preferred ROR-provided display name.",
+    )
+    label_for = models.ForeignKey(
+        'Organization',
+        on_delete=models.CASCADE,
+        related_name='labels',
+        blank=True,
+        null=True,
+        help_text="This name is a preferred ROR-provided alternative, "
+            "often in a different language from the ROR display name.",
+    )
+    custom_label_for = models.OneToOneField(
+        'Organization',
+        on_delete=models.CASCADE,
+        related_name='custom_label',
+        blank=True,
+        null=True,
+        help_text="This name is a custom label entered by the end user. "
+            "Only exists in Janeway, independent of ROR.",
+    )
+    alias_for = models.ForeignKey(
+        'Organization',
+        on_delete=models.CASCADE,
+        related_name='aliases',
+        blank=True,
+        null=True,
+        help_text="This name is a less preferred ROR-provided alternative.",
+    )
+    acronym_for = models.ForeignKey(
+        'Organization',
+        on_delete=models.CASCADE,
+        related_name='acronyms',
+        blank=True,
+        null=True,
+        help_text="This name is a ROR-provided acronym.",
+    )
+    language = models.CharField(
+        max_length=10,
+        blank=True,
+        choices=submission_models.LANGUAGE_CHOICES,
+    )
+
+    def __str__(self):
+        return self.value
+
+
+def validate_ror_id(ror_id):
+    ror_regex = '^0[a-hj-km-np-tv-z|0-9]{6}[0-9]{2}$'
```

Comment:
It is good practice to [compile the regex](https://docs.python.org/3/library/re.html#re.compile) explicitly as a constant, both for performance and reusability. Also, remember to flag the string as a regex with `r'^0[a-hj-km-np-tv-z|0-9]{6}[0-9]{2}$'`